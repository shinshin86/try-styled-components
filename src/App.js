import React, { Component } from 'react';
import logo from './logo.svg';
import styled, { css, keyframes } from 'styled-components';
import CustomForm from './CustomForm';

/*** styled components ***/
const AppBody = styled.div`
  text-align: center;
`;

// should define beore AppLogo
const AppLogoSpin = keyframes`
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
`;

const AppLogo = styled.img`
  animation: ${AppLogoSpin} infinite 20s linear;
  height: 80px;
`;

const AppHeader = styled.header`
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
`;

const AppTitle = styled.h1`
  font-size: 1.5em;
`;

const AppIntro = styled.p`
  font-size: large;
`;

const Button = styled.button`
  border-radius: 3px;
  padding: 0.25em 1em;
  margin: 0 1em;
  background: transparent;
  color: palevioletred;
  border: 2px solid palevioletred;

  ${props =>
    props.primary &&
    css`
      background: palevioletred;
      color: white;
    `} ${props =>
    props.large &&
    css`
      border: 5px solid palevioletred;
      font-size: 26px;
    `};
`;

class App extends Component {
  render() {
    return (
      <AppBody>
        <AppHeader>
          <AppLogo src={logo} alt="logo" />
          <AppTitle>Try styled components with react create app</AppTitle>
        </AppHeader>
        <AppIntro>
          <Button>Normal Button</Button>
          <Button primary>primary button</Button>
          <Button primary large>
            large primary button
          </Button>
        </AppIntro>
        <CustomForm />
      </AppBody>
    );
  }
}

export default App;
