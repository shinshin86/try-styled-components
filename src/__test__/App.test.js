import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import App from '../App';

test('App Snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
