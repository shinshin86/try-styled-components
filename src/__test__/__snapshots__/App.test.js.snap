// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot 1`] = `
.c15 {
  color: palevioletred;
  font-size: 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
  type: text;
  margin: 1em;
  padding: 1em;
}

.c16 {
  color: palevioletred;
  font-size: 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
  type: date;
  margin: 1em;
  padding: 1em;
}

.c17 {
  color: palevioletred;
  font-size: 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
  type: password;
  margin: 1em;
  padding: 1em;
}

.c19 {
  color: palevioletred;
  font-size: 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
  type: image;
  margin: 12px;
  padding: 12px;
}

.c20 {
  color: palevioletred;
  font-size: 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
  type: submit;
  margin: 1em;
  padding: 1em;
}

.c18 {
  margin: 16px;
  padding: 16px;
  background-color: grey;
}

.c14 {
  border-bottom: solid 1px palevioletred;
}

.c13 {
  border: solid 1px grey;
  margin: 16px;
  padding: 16px;
}

.c9 {
  margin: 30px;
  padding: 30px;
  border: solid 1px #ccc;
}

.c11 {
  margin: 100px;
  border: solid 1px #ccc;
}

.c10 {
  background: #134a63;
  color: #eee;
  width: 300px;
  height: 50px;
  margin: 30px;
}

.c12 {
  color: #de5;
  width: 250px;
  height: 70px;
}

.c0 {
  text-align: center;
}

.c4 {
  max-width: 1100px;
  margin: 0 auto;
}

.c2 {
  -webkit-animation: iVXCSc infinite 20s linear;
  animation: iVXCSc infinite 20s linear;
  height: 80px;
}

.c1 {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.c3 {
  font-size: 1.5em;
}

.c5 {
  font-size: large;
}

.c6 {
  border-radius: 3px;
  padding: 0.25em 1em;
  margin: 0 1em;
  background: transparent;
  color: palevioletred;
  border: 2px solid palevioletred;
}

.c7 {
  border-radius: 3px;
  padding: 0.25em 1em;
  margin: 0 1em;
  background: transparent;
  color: palevioletred;
  border: 2px solid palevioletred;
  background: palevioletred;
  color: white;
}

.c8 {
  border-radius: 3px;
  padding: 0.25em 1em;
  margin: 0 1em;
  background: transparent;
  color: palevioletred;
  border: 2px solid palevioletred;
  background: palevioletred;
  color: white;
  border: 5px solid palevioletred;
  font-size: 26px;
}

<div
  className="c0"
>
  <header
    className="c1"
  >
    <img
      alt="logo"
      className="c2"
      src="logo.svg"
    />
    <h1
      className="c3"
    >
      Try styled components with react create app
    </h1>
  </header>
  <div
    className="c4"
  >
    <p
      className="c5"
    >
      <button
        className="c6"
      >
        Normal Button
      </button>
      <button
        className="c7"
      >
        primary button
      </button>
      <button
        className="c8"
      >
        large primary button
      </button>
    </p>
    <div
      className="c9"
    >
      <button
        className="c10"
        onClick={[Function]}
      >
        Toggle Button
      </button>
      <div
        className="c11"
      >
        <button
          className="c12 c10"
          onClick={[Function]}
        >
          Toggle Button
        </button>
      </div>
    </div>
    <div
      className="c13"
    >
      <h2
        className="c14"
      >
        Custom Form
      </h2>
      <input
        className="c15"
        margin="1em"
        padding="1em"
        placeholder="text is here"
        type="text"
      />
      <input
        className="c16"
        margin="1em"
        padding="1em"
        placeholder="date is here"
        type="date"
      />
      <input
        className="c17"
        margin="1em"
        padding="1em"
        placeholder="password is here"
        type="password"
      />
      <p
        className="c18"
      >
        <input
          alt="react logo"
          className="c19"
          height="200"
          margin="12px"
          padding="12px"
          placeholder="password is here"
          size="12px"
          src="logo.svg"
          type="image"
          width="200"
        />
      </p>
      <input
        className="c20"
        margin="1em"
        padding="1em"
        type="submit"
        value="submit"
      />
    </div>
  </div>
</div>
`;
